### PR TITLE
Make virtual mode an option using msp backup and restore

### PR DESCRIPTION
**Pull request by haslinghuis**
After some confusion in https://github.com/betaflight/betaflight-configurator/pull/2742 we decided to:

- [x] Remove virtual connection from PortHandler. [1]
- [x] Make virtual connection an option as most people don't use this feature. [1]
- [x] Restore backup and restore if option is enabled to be able to backup a real target and restore on a virtual one.
- [x] Remove restore button in serial mode from firmware >= 4.3 if virtual option enabled.
- [x] Move showAllSerialDevices to PortHandler for optimization. 

[1] _please re-enable Virtual Mode in Options tab if needed_.

image

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by Asizon** [comment]
Wow thankyou for the changes @haslinghuis 
Maybe we should put virtual mode in the button?some like "Backup(Only VirtualMode)" or similar.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

File: locales/en/messages.json
Comment (McGiverGim) [inline locales/en/messages.json:114]: This description is only for the translators. It must "describe" what are they seeing to help them translate them.

Comment (haslinghuis) [reply locales/en/messages.json]: Fixed!

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by limonspb** [comment]
Speaking of Backup and Restore buttons:
1. Keeping these buttons as is for 4.2 means we will keep misleading 4.2 users.
They are broken not just in 4.3 but also in 4.2 and who knows about 4.1.
For example in 4.2 they are not restoring OSD, Power tab, Vtx table, filters, advanced PID parameters **and who knows what else**.

2. in 4.3 they are even more broken.

3. HOWEVER i can see why some testers still like to use these buttons.

Considering 1.2.3 we need to keep these buttons but make sure only testers and people that know what they are doing are hitting these buttons.

I am proposing to rename these buttons to "Save Json, Load Json" and moving them into a separate "Testing" panel, something like that:
image
that way the current functionality (broken or not) is still there for testers, but it is not confusing regular users anymore.

**Review by limonspb — APPROVED** [review]

**Review by limonspb — APPROVED** [review]

**Review by ctzsnooze — APPROVED** [review]

**Review by Asizon — APPROVED** [review]
Lgtm! Thankyou @haslinghuis

**Pull request merged**
